src/TerminalFighter/Settings/Settings.cpp

```diff
@@ -1,44 +1,232 @@
+    } catch (YAML::InvalidNode e) {
+        LOG(ERROR) << "Could not load setting " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << ", " << e.msg;
+        return false;
+    } catch (YAML::RepresentationException e) {
+        LOG(ERROR) << "Couldn't load " << vec_to_str(keys) << " as string. " << e.msg;
+        return false;
+    }
+
+    return true;
+}
+
+bool Settings::load_int(
+    SettingsSection section,
+    std::vector<std::string> keys,
+    int& value) const {
+
+    YAML::Node node = load_node(section, keys);
+
+    if (!node.IsScalar()) {
+        LOG(ERROR) << "SettingsGroup " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << " is not int";
+        return false;
     }
 
     try {
-        asset_paths_ = YAML::LoadFile(ASSETS_PATHS_FILE);
-        LOG(INFO) << "Successfully loaded " << ASSETS_PATHS_FILE;
-    } catch (YAML::BadFile) {
-        LOG(ERROR) << "Failed to load file: " << ASSETS_PATHS_FILE;
-        errors++;
+        value = node.as<int>();
+    } catch (YAML::InvalidNode e) {
+        LOG(ERROR) << "Could not load setting " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << ", " << e.msg;
+        return false;
+    } catch (YAML::RepresentationException e) {
+        LOG(ERROR) << "Couldn't load " << vec_to_str(keys) << " as int. " << e.msg;
+        return false;
     }
 
-    return !errors;
+    return true;
 }
 
-const YAML::Node& Settings::video_settings() const {
-    return video_settings_;
+bool Settings::load_double(
+    SettingsSection section,
+    std::vector<std::string> keys,
+    double& value) const {
+
+    YAML::Node node = load_node(section, keys);
+
+    if (!node.IsScalar()) {
+        LOG(ERROR) << "SettingsGroup " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << " is not double";
+        return false;
+    }
+
+    try {
+        value = node.as<double>();
+    } catch (YAML::InvalidNode e) {
+        LOG(ERROR) << "Could not load setting " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << ", " << e.msg;
+        return false;
+    } catch (YAML::RepresentationException e) {
+        LOG(ERROR) << "Couldn't load " << vec_to_str(keys) << " as double. " << e.msg;
+        return false;
+    }
+
+    return true;
 }
 
-const YAML::Node& Settings::asset_paths() const {
-    return asset_paths_;
+bool Settings::load_bool(
+    SettingsSection section,
+    std::vector<std::string> keys,
+    bool& value) const {
+
+    YAML::Node node = load_node(section, keys);
+
+    if (!node.IsScalar()) {
+        LOG(ERROR) << "SettingsGroup " << vec_to_str(keys)
+                   << " from section " << static_cast<char>(section)
+                   << " is not boolean";
```

Comment:
I understand that this message is "technically" true, but it still feels weird to me XD. 
Technically, a `string` or an `int` can get by here since they're also `scalar`. 

But I honestly have like no good suggestions so this is probs the way to go atm lol